llvm/include/llvm/Support/RISCVAttributes.h

```diff
@@ -32,8 +32,21 @@ enum AttrType : unsigned {
   PRIV_SPEC = 8,
   PRIV_SPEC_MINOR = 10,
   PRIV_SPEC_REVISION = 12,
+  ATOMIC_ABI = 14,
 };
 
+namespace RISCVAtomicAbiTag {
```

Comment:
scoped enum (`enum class`) is better than namespace+unscoped enum

Reply:
I've followed your advice here, but this seems to require quite a lot of `static_casting` when what we actually want is a typed alias for an integer which is precisely what a standard enum type is. I've also had to introduce what I'd say are useless cases to the switch for unreachable types we've already checked.